hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -3186,7 +3187,8 @@ public String getCanonicalServiceName() {
     entryPoint(Statistic.INVOCATION_GET_DELEGATION_TOKEN);
     LOG.debug("Delegation token requested");
     if (delegationTokens.isPresent()) {
-      return delegationTokens.get().getBoundOrNewDT(encryptionSecrets);
+      return delegationTokens.get().getBoundOrNewDT(encryptionSecrets,
+          (renewer!=null ? new Text(renewer) : null));
```

Comment:
Is it possible for renewer to not be null but actually be empty?

Reply:
It is possible for it to be empty, in which case the renewer Text object will be "empty", and that case is checked by yarn.

Reply:
* need to make sure that null results in renewer == new Text()
* add some spaces round the `!=`